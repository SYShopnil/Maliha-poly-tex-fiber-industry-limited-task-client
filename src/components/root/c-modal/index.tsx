"use client";
import React, { useState, useEffect, ReactNode } from "react";

interface IModal {
  show: boolean;
  onClose: () => void;
  children: ReactNode;
}

export const CModal = ({ show, onClose, children }: IModal) => {
  const [isVisible, setIsVisible] = useState(show);
  const [shouldAnimate, setShouldAnimate] = useState(false);
  // console.log(`render outside`);
  // console.log({ shouldAnimate, isVisible, show });
  useEffect(() => {
    if (show) {
      setIsVisible(true);
      setTimeout(() => setShouldAnimate(true), 200); // Duration of the fade-in animation
      //   setShouldAnimate(true);
    } else {
      setTimeout(() => setShouldAnimate(false), 200); // Duration of the fade-out animation
      setTimeout(() => setIsVisible(false), 200); // Duration of the fade-out animation
    }
  }, [show]);

  if (!isVisible) return null;

  return (
    <div
      className={`fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center transition-opacity duration-500 ${
        show && shouldAnimate ? "opacity-100" : "opacity-0"
      }`}
      onClick={onClose}
    >
      <div
        className="bg-white p-4 rounded shadow-lg transition-transform duration-300 overflow-x-hidden overflow-y-scroll max-h-[80vh] w-[65%]"
        onClick={(e) => e.stopPropagation()} // Prevent event propagation to avoid closing modal
      >
        {children}
      </div>
    </div>
  );
};
