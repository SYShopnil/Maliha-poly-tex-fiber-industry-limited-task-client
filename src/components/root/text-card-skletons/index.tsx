import React from "react";

export const STextCardSkelton = () => {
  const counter = new Array(9).fill(0);
  return (
    <div className={`grid grid-cols-12 gap-2`}>
      {counter.map((_, ind) => {
        return (
          <div
            className={`col-span-12 md:col-span-6 lg:col-span-4  shadow-lg p-3 `}
            key={ind}
          >
            <>
              <div className="text-blue-600 font-bold mb-2 bg-gray-200 h-6 rounded w-1/2 "></div>
              <div className="text-gray-700 mb-4 bg-gray-200 h-4 rounded w-3/4"></div>
              <div className="flex justify-between">
                <button className="bg-gray-200 w-8 text-gray-600 py-2 px-4 rounded-md cursor-not-allowed"></button>
                <button className="bg-gray-200 w-8  text-gray-600 py-2 px-4 rounded-md cursor-not-allowed"></button>
              </div>
            </>
          </div>
        );
      })}
    </div>
  );
};
