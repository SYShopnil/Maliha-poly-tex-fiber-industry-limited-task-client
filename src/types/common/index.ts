export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  SHeaderMain = "SHeaderMain",
  SProfile = "SProfile",
}
