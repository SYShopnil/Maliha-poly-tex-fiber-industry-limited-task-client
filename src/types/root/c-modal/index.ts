import { ReactNode } from "react";

export interface IModal {
  show: boolean;
  onClose: () => void;
  children: ReactNode;
}
