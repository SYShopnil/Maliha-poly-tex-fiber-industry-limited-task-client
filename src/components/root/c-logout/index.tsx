"use client";
import React from "react";
import { Button } from "../button";
import { BtnColorSchema } from "@src/types/root";
import { logoutController } from "@src/lib/login-handler";
import { EDataTestId } from "@src/types/common";

const CLogout = () => {
  const logoutHandler = async () => {
    try {
      await logoutController();
    } catch (err) {
      console.log(err);
    }
  };
  return (
    <div role={EDataTestId.CLogout}>
      <Button
        btnText="Logout"
        colorSchema={BtnColorSchema.SolidBgVioletTextWhite}
        isArrow={false}
        clickHandler={logoutHandler}
      />
    </div>
  );
};
export default CLogout;
