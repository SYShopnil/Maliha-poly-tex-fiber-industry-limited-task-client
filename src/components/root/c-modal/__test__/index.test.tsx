import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { CModal } from "..";
import { IModal } from "@src/types/root/c-modal";

const mock_data: IModal = {
  show: true,
  onClose: () => {},
  children: <h1>Bye world</h1>,
};

describe("When Modal Component Has Rendered", () => {
  it("Expect render successfully", () => {
    render(
      <CModal {...mock_data}>
        <h1>Hello world</h1>
      </CModal>
    );
    const myElement = screen.getByRole(EDataTestId.CModal);
    expect(myElement).toMatchSnapshot();
  });
});
