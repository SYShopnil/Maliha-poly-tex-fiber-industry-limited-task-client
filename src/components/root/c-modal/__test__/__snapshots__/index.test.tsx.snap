// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Modal Component Has Rendered Expect render successfully 1`] = `
<div
  class="fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center transition-opacity duration-500 opacity-0"
  role="CModal"
>
  <div
    class="bg-white p-4 rounded shadow-lg transition-transform duration-300 overflow-x-hidden overflow-y-scroll max-h-[80vh] w-[65%]"
  >
    <h1>
      Hello world
    </h1>
  </div>
</div>
`;
